rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -748,6 +749,74 @@ private void logCommitmentWithVotedFederation(BridgeEventLogger eventLogger, Fed
+        if (!proposedFederationOpt.isPresent()) {
+            String message = "Proposed federation should be present when creating SVP fund transaction.";
+            logger.warn(message);
+            throw new IllegalStateException(message);
+        }
+        Federation proposedFederation = proposedFederationOpt.get();
+
+        BtcTransaction svpFundTransactionUnsigned = new BtcTransaction(constants.getBtcParams());
+        svpFundTransactionUnsigned.setVersion(BTC_TX_VERSION_2);
+
+        addOutputToProposedFederation(svpFundTransactionUnsigned, proposedFederation, feePerKb);
+
+        // complete tx with input and change output
+        SendRequest sendRequest = createSendRequest(svpFundTransactionUnsigned, feePerKb);
+        activeFederationWallet.completeTx(sendRequest);
+
+        provider.setSvpFundTxHashUnsigned(svpFundTransactionUnsigned.getHash());
+        return svpFundTransactionUnsigned;
+    }
+
+    private void addOutputToProposedFederation(BtcTransaction svpFundTransaction, Federation proposedFederation, Coin feePerKb) {
+        long transactionSizeToBeSentFromProposedFederation = calculateTransactionSizeToBeSentFromFederation(proposedFederation);
+        Coin neededValueToSendFromProposedFederation = feePerKb.multiply(transactionSizeToBeSentFromProposedFederation);
+
+        svpFundTransaction.addOutput(neededValueToSendFromProposedFederation, proposedFederation.getAddress());
+    }
+
+    private long calculateTransactionSizeToBeSentFromFederation(Federation federation) {
+        long inputInfo = 36;
+        long outputInfo = 32;
+
+        long scriptSigSize = getFederationScriptSigSize(federation);
+        long txBaseSize = inputInfo + outputInfo + scriptSigSize;
+        long txTotalSize = inputInfo + outputInfo + getFederationTransactionWitnessSize(federation);
+
+        return (3 * txBaseSize + txTotalSize) / 4;
+    }
+
+    private int getFederationScriptSigSize(Federation federation) {
+/*        if (federation.isWrappedSegwit()) // TODO {
+            return 36;
+        }
+        if (federation.isSegwit()) {
+            return 0;
+        }*/
+
+        return getFederationTransactionWitnessSize(federation);
+    }
+
+    private int getFederationTransactionWitnessSize(Federation federation) {
+        int redeemScriptSize = federation.getRedeemScript().getProgram().length;
+        int neededSignaturesSize = 72 * federation.getNumberOfSignaturesRequired();
+        return redeemScriptSize + neededSignaturesSize;
+    }
+
+    private SendRequest createSendRequest(BtcTransaction transaction, Coin feePerKb) {
+        SendRequest sendRequest = SendRequest.forTx(transaction);
+        sendRequest.changeAddress = getActiveFederationAddress();
+        sendRequest.feePerKb = feePerKb;
+        sendRequest.missingSigsMode = Wallet.MissingSigsMode.USE_OP_ZERO;
+        sendRequest.recipientsPayFees = false;
```

Comment:
we dont want the proposed fed to pay for the fees since the amount to send was calculated quite precise

Reply:
Interesting, this way no matter how high the current network fees are we ensure the proposed will receive enough funds to send back, right?

Let's take a note to double check this anyway

Reply:
i think that's always a risk, because the fees could vary between the fund transaction and the spend transaction are sent. But I wanted to make sure that the value chosen to be the one received from the proposed fed will actually be the value sent to it